docs/source/howto/query.rst

```diff
@@ -90,6 +94,14 @@ For example, you can iterate over the results of your query in a for loop:
     for entry in qb.iterall():
         # do something with a single entry in the query result
 
+.. important::
+
+    When the query result is expected to be large, do not use ``dict`` and ``all`` but use the generator variants ``iterdict`` and ``iterall``.
+    The ``dict`` and ``all`` methods load the entire query result in memory which can lead to out-of-memory errors.
+    It also makes updating the data very slow as each change results in a single database commit.
+    The ``iterdict`` and ``iterall`` generators open a database transaction, committing all changes only once at the end of the generator.
```

Comment:
Here my suggestion
```suggestion
    When looping over the result of a query, use the ``iterall`` (or ``iterdict``) generators instead of ``all`` (or ``dict``).
    This avoids loading the entire query result into memory, and it also delays committing changes made to AiiDA objects inside the loop until the end of the loop is reached.
```